File: airesources/Java/hlt/DockMove.java
package hlt;

public class DockMove extends Move {

    private final long destinationId;

    public DockMove(final Ship ship, final Planet planet) {
        super(MoveType.Dock, ship);
        destinationId = planet.getId();
    }

    public long getDestinationId() {
        return destinationId;
    }
}

File: airesources/CSharp/Halite2/hlt/DockMove.cs
namespace Halite2.hlt
{

    public class DockMove : Move
    {

        private long destinationId;

        public DockMove(Ship ship, Planet planet)
            : base(MoveType.Dock, ship)
        {
            destinationId = planet.GetId();
        }

        public long GetDestinationId()
        {
            return destinationId;
        }
    }
}